File: src/main/java/bilibili/lingaly/tutorial/item/ModFoodComponent.java
package bilibili.lingaly.tutorial.item;

import net.minecraft.entity.effect.StatusEffectInstance;
import net.minecraft.entity.effect.StatusEffects;
import net.minecraft.item.FoodComponent;

public class ModFoodComponent {

    public static FoodComponent fooda = new FoodComponent.Builder()
            .hunger(6)
            .saturationModifier(0.8f)
            .statusEffect(new StatusEffectInstance(StatusEffects.LUCK, 1000),
                    1.0f)
            .alwaysEdible()
            .build();

}

File: src/main/kotlin/bilibili/lingaly/ktmccouse/items/ModFoodComponent.kt
package bilibili.lingaly.ktmccouse.items

import net.minecraft.entity.effect.StatusEffect
import net.minecraft.entity.effect.StatusEffectInstance
import net.minecraft.entity.effect.StatusEffects
import net.minecraft.item.FoodComponent

object ModFoodComponent {

    val fooda: FoodComponent = FoodComponent
        .Builder()
        .hunger(6)
        .saturationModifier(0.8f)
        .statusEffect(
            StatusEffectInstance(
                StatusEffects.LUCK,
                1000
            ),
            1.0f
        )
        .alwaysEdible()
        .build()

}